### PR TITLE
Fix phase panel

### LINKED ISSUE #53: Add test for phasing panel
### Description of the bug

Most of the small tests we use have `phased = true`. A test should be added that evaluates the output of the `VCF_PHASE_PANEL` subworkflow. Specifically this part:

```
    if (params.phased == false) {
        VCF_PHASE_SHAPEIT5(ch_vcf
            .map { meta, vcf, csi -> [meta, vcf, csi, [], meta.region] },
        Channel.of([[],[],[]]).collect(),
        Channel.of([[],[],[]]).collect(),
        Channel.of([[],[]]).collect())
        ch_versions = ch_versions.mix(VCF_PHASE_SHAPEIT5.out.versions)
        ch_panel_phased = VCF_PHASE_SHAPEIT5.out.variants_phased
            .combine(VCF_PHASE_SHAPEIT5.out.variants_index, by: 0)
    } else {
        ch_panel_phased = ch_vcf
    }
```

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by LouisLeNezet**
Fix #53 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phaseimpute/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phaseimpute _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by LouisLeNezet** [comment]
Need PR #65 to pass the test.

File: subworkflows/local/vcf_phase_shapeit5/main.nf [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    main:
+
+    ch_versions = Channel.empty()
+
+    // It is needed to generate a file containing the region to phase in a Chr \tab Start \tab End format
+
+    // Create the File in bed format and use the meta id for the file name
+    ch_region_file = ch_region
+        .collectFile(newLine: true) { metaCR, region -> ["${metaCR.chr}.bed", region.replace(":","\t").replace("-","\t")]}
+        .map { file -> [[id: file.getBaseName(), chr:file.getBaseName()], file] }
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply subworkflows/local/vcf_phase_shapeit5/main.nf:25]: If input region is all then the ch_region should be generated correctly to `[[chr, region], region]`
Testing with an `all` option wouldn't work in this subworkflow.

File: conf/test_panelprep.config [hunk 2]
```diff
@@ -20,9 +20,10 @@ params {
     max_time   = '1.h'
 
     // Genome references
-    fasta  = params.pipelines_testdata_base_path + "reference_genome/21_22/hs38DH.chr21_22.fa"
-    panel  = "${projectDir}/tests/csv/panel.csv"
-    phased = true
+    fasta        = params.pipelines_testdata_base_path + "reference_genome/21_22/hs38DH.chr21_22.fa"
+    input_region = "${projectDir}/tests/csv/region.csv"
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/test_panelprep.config]: True, the input_region could be `all` as it therefore extract all the regions from the fasta files.
I've modified it for the test.

**Comment by LouisLeNezet** [comment]
@atrigila Unfortunately for the moment we cannot use the "all" parameter as it not supported by nf-validation.
When we will move to nf-schema it will be possible to do so.

File: conf/test_panelprep.config (hunk 2 above)
[the remark this answers is not in the document]
Comment (atrigila) [reply conf/test_panelprep.config:24]: I was thinking that when `params.input_region` is `null`, the region should default to the full chromosome. To ensure our implementation handles this correctly, it would be beneficial to add a test case where `input_region` is not specified. This way, we can confirm that the code runs as expected for full chromosomes.

[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/test_panelprep.config:24]: Okay I'll try !

[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/test_panelprep.config:24]: So, it cannot be done as the fasta file contains the whole chr21 and chr22.
Therefore the regions are chr21:0-46709983 and chr22:0-50818468.
For a subset of test the fasta files would need to be of the regions of interest.

But if we subset the fasta file to a subregion using `samtools faidx` then all the tools using the fasta file will crash as the string name of the contigs will be `chr22:16570000-16610000` instead of `chr22`. I've tried with the `params.pipelines_testdata_base_path + "reference_genome/21_22/hs38DH.chr21_22.s.fa"`

There is no good solution from my point of view.
What I would suggest is:
- if you don't want to specify a region, then it will run at least on full contigs names provide by fasta (this cannot be done with the test datasets we have for the moment, but this should still work).
- if you want to run on a specific region, fill the csv region file.

[the remark this answers is not in the document]
Comment (atrigila) [reply conf/test_panelprep.config:24]: I understand, it is something we'll have to take this into account. We can add "medium-sized" test datasets (e.g. the ref panel for a single full chromosome) to inspect these edge cases before we run the full-size test. We use a subset of the full-size test, such as:

```
panel,chr,vcf,index
1000GP.s.norel,chr22,http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20201028_3202_phased/CCDG_14151_B01_GRM_WGS_2020-08-05_chr22.filtered.shapeit2-duohmm-phased.vcf.gz,http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20201028_3202_phased/CCDG_14151_B01_GRM_WGS_2020-08-05_chr22.filtered.shapeit2-duohmm-phased.vcf.gz.tbi
```
Do you think this is feasible?

[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/test_panelprep.config:24]: This could be done but this can't be used for the github CI as it would be too much data to handle I think.
But for the test_full we will for sure use this kind of data !

[the remark this answers is not in the document]
Comment (atrigila) [reply conf/test_panelprep.config:24]: Indeed, we don’t have to include all tests in the GitHub CI environment, especially when dealing with such big data files. 

I tested the medium-sized dataset locally (it took some time to download the files), but the process broke. This suggests we need to further explore what happens with phasing in full chromosomes.

[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/test_panelprep.config:24]: I'll try that and fix it next week !

**Review by atrigila — APPROVED** [review]
Good job :) we have just to add a new issue taking into account that we should test this new functionality in a bigger dataset.
